gradle-diff-task-dependencies/sample/build.gradle

```diff
@@ -0,0 +1,27 @@
+import com.novoda.gradle.diffdependencytask.DiffDependencyTaskPlugin
+
+apply plugin: 'java'
+apply plugin: DiffDependencyTaskPlugin
+
+diffDependency {
+    // custom changed files provider available to test in non-CI environments
+    changedFilesProvider = {
+        return ['apple/fastlane', 'android/build.gradle']
+    }
+}
+
+task('testAndroid') << {
+    println 'This is an android test task.'
+}
+
+task('testApple') << {
+    println 'This is an apple test task.'
+}
+
+task('selectiveTest') << {
+    println 'This will run after 0 or more between "testAndroid" and "testApple".'
+}
+
+tasks.getByName('selectiveTest')
+        .ifDiffMatches(~'android/.*').dependsOn('testAndroid')
```

Comment:
I have to say I'm not big fan of this API: why do you need to break the chain in 2 steps (`ifDiffMatches()` + `dependsOn`)? Is `ifDiffMatches()` considered a standalone operation? To me it seems like you always have to complete via `dependsOn()` (that by the way might be confusing if you're used to `Task#dependsOn`), so why not having a single call with both parameters? You could use varargs for tasks if you want to add multiple conditional depedencies